thunder.post_update.php

```diff
@@ -22,3 +24,81 @@ function thunder_post_update_ensure_config_selector_installed() {
   // Output logged messages to related channel of update execution.
   return $thunderUpdater->logger()->output();
 }
+
+/**
+ * Installs the content_content_lock view.
+ */
+function thunder_post_update_add_content_lock_view() {
+
+  $content_view = View::load('content');
+  if (empty($content_view)) {
+    return;
+  }
+  $content_view->setThirdPartySetting('config_selector', 'feature', 'thunder_content_view');
+  $content_view->setThirdPartySetting('config_selector', 'priority', 0);
+
+  $view_contains_content_lock = FALSE;
+  foreach (array_keys($content_view->get('display')) as $display_name) {
+    $display = $content_view->getDisplay($display_name);
+    $options = ['fields', 'filters', 'sorts', 'relationships'];
+    foreach ($options as $option) {
+      if (!empty($display['display_options'][$option]) && is_array($display['display_options'][$option])) {
+        foreach ($display['display_options'][$option] as $key => $values) {
+          if ($values['table'] == 'content_lock') {
+            $view_contains_content_lock = TRUE;
+          }
+        }
+      }
+    }
+  }
+
+  if ($view_contains_content_lock) {
+    // Duplicate content into content_content_lock view and add third party
+    // settings.
+    $content_lock_view = $content_view->createDuplicate();
+    $content_lock_view->set('id', 'content_content_lock');
+    $content_lock_view->setThirdPartySetting('config_selector', 'feature', 'thunder_content_view');
```

Comment:
Not sure we need reset the feature. I think it might clearer without this.

Reply:
Its not actually resetting, its adding the third party settings to the original view, that did not have it.

Reply:
It does though - we're adding them above.

Reply:
Yeah, I just missed line 37.